cmd/web-handlers.go

```diff
@@ -2525,37 +2533,57 @@ func (d *DealVo) setDefault() {
 	if len(d.Price) == 0 {
 		d.Price = "0"
 	}
-	if len(d.SwanEndpoint) == 0 {
-		d.SwanEndpoint = os.Getenv("SWAN_API")
-		if len(strings.TrimSpace(d.SwanEndpoint)) == 0 {
-			d.SwanEndpoint = "https://api.filswan.com"
-		}
-		os.Setenv("SWAN_API", d.SwanEndpoint)
-	} else {
-		os.Setenv("SWAN_API", d.SwanEndpoint)
-	}
-	if len(d.SwanApiToken) == 0 {
-		d.SwanApiToken = os.Getenv("SWAN_TOKEN")
-	} else {
-		os.Setenv("SWAN_TOKEN", d.SwanApiToken)
-	}
+	//if len(d.SwanEndpoint) == 0 {
+		//d.SwanEndpoint = os.Getenv("SWAN_API")
+		//if len(strings.TrimSpace(d.SwanEndpoint)) == 0 {
+			//d.SwanEndpoint = "https://api.filswan.com"
+		//}
+		//os.Setenv("SWAN_API", d.SwanEndpoint)
+	//} else {
+		//os.Setenv("SWAN_API", d.SwanEndpoint)
+	//}
+	//if len(d.SwanApiToken) == 0 {
+		//d.SwanApiToken = os.Getenv("SWAN_TOKEN")
+	//} else {
+		//os.Setenv("SWAN_TOKEN", d.SwanApiToken)
+	//}
 	if len(d.MinerId) == 0 {
 		d.MinerId = "f0447183"
 	}
 }
 
+
+func ExecCommand(strCommand string) (string, error) {
+	cmd := exec.Command("/bin/bash", "-c", strCommand)
+	stdout, _ := cmd.StdoutPipe()
+	if err := cmd.Start(); err != nil {
+		fmt.Println("Execute failed when Start:" + err.Error())
+		return "", err
+	}
+	out_bytes, _ := ioioutil.ReadAll(stdout)
+	if err := stdout.Close(); err != nil {
+		fmt.Println("Execute failed when close stdout:" + err.Error())
+		return "", err
+	}
+	if err := cmd.Wait(); err != nil {
+		fmt.Println("Execute failed when Wait:" + err.Error())
+		//return "", err
+	}
+	return string(out_bytes), nil
+}
+
+
 // SendDeal - send deal to filecoin network.
 func (web *webAPIHandlers) SendDeal(w http.ResponseWriter, r *http.Request) {
 
 	decoder := json.NewDecoder(r.Body)
-	var dealVo DealVo
-	err := decoder.Decode(&dealVo)
+	var onlineDealRequest OnlineDealRequest
+	err := decoder.Decode(&onlineDealRequest)
 	if err != nil && err != io.EOF {
 		w.Write([]byte(fmt.Sprintf("bad request: %s", err.Error())))
 		return
 	}
-	dealVo.setDefault()
-
+	fmt.Println("hello1")
```

Comment:
remove:print hello

Reply:
already removed. Maybe the code line is not updated by github.